src/core/endpoint-parser/ApiDocFieldsParser.ts

```diff
@@ -84,31 +88,31 @@ export class ApiDocFieldsParser {
     }
 
     private createObjectProperties(parentProperties: JsonSchema, currentNamePart: string) {
+        // eslint-disable-next-line no-param-reassign
         parentProperties[currentNamePart] = parentProperties[currentNamePart] || {
             type: "object",
         };
+        // eslint-disable-next-line no-param-reassign
         parentProperties[currentNamePart].properties = parentProperties[currentNamePart].properties || {};
         return parentProperties[currentNamePart].properties;
     }
 
     private createArrayProperties(parentProperties: JsonSchema, currentNamePart: string) {
-        parentProperties[currentNamePart].items =
-            parentProperties[currentNamePart].items || {
-                type: "object",
-            };
+        // eslint-disable-next-line no-param-reassign
```

Comment:
maybe let's remove this rule at all?

Reply:
I actually like this rule, it forces you to write a much more maintainable code. Why do you think we should remove it?

Reply:
I just see too much `// eslint-disable-next-line no-param-reassign` in this file
Shall we refactor this code so we don't have to disable linter in these places?
I guess we can create separate issue to refactor code(please do) and implement it as a separate pull request.

Reply:
Yep, I was just about to create issues